app/assets/javascripts/discourse/app/lib/discourse-location.js

```diff
@@ -2,21 +2,12 @@ import EmberObject from "@ember/object";
 import { defaultHomepage } from "discourse/lib/utilities";
 import { guidFor } from "@ember/object/internals";
 import { withoutPrefix } from "discourse-common/lib/get-url";
+import uniqueId from "discourse/helpers/unique-id";
+
 let popstateFired = false;
 const supportsHistoryState = window.history && "state" in window.history;
 const popstateCallbacks = [];
 
-function _uuid() {
-  return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function (c) {
-    let r, v;
-    /* eslint-disable no-bitwise */
-    r = (Math.random() * 16) | 0;
-    v = c === "x" ? r : (r & 3) | 8;
-    /* eslint-enable no-bitwise */
-    return v.toString(16);
-  });
-}
-
```

Comment:
Is it ok if I did this? (moving this to a uniqueId helper)

Reply:
Fine to copy/paste to a helper, but please don't change `discourse-location.js`. It comes from Ember's own [history-location](https://github.com/emberjs/ember.js/blob/9e51c3629d18f53bfc057ac94987b5d27eb7cdfa/packages/%40ember/routing/history-location.ts#L12-L19) implementation, which we hope to switch to very soon, so I'd rather our implementation didn't diverge any further.

Reply:
Noting this is also super similar to https://github.com/discourse/discourse/blob/8ef569cda4b95029922ecbc348ee8aaca9152fc6/app/assets/javascripts/pretty-text/addon/guid.js which I talked about in another comment...